libproject/src/libproject.cpp

```diff
@@ -69,10 +66,10 @@ namespace Interface {
         /*virtual*/ bool open();
 
         /*!
-         * \brief Gives to user list<string> of filenames
-         * \return list<string> of filenames of project or drops exception if project wasn't loaded
+         * \brief Gives to user filepaths of files
+         * \return return collection of files of project, in case of failure throws exception
          */
-        /*virtual*/ const list<string> getFileNames() const;
+        /*virtual*/ const FileSetLoader::Files getFileNames() const;
```

Comment:
You don't need to make full qualified name in declaration. It's not an error, but in header file signature cant looks just like:
```cpp
struct Foo {
    typedef Foo* FooP;
    static FooP create();
};
//but in declaration
Foo::FooP Foo::create() {
    return new Foo();
}
// and in usage:
void main() {
Foo::FooP foo = Foo::create();
} 
```